From: Gergely Nagy <algernon@madhouse-project.org>
Date: Fri, 21 Dec 2012 16:07:43 +0100
Subject: playercode/mdreg: Register the NULL driver before the file writers

Register the NULL driver sooner, as having no sound is preferable to
writing to music.raw: file writing should be used only when explicitly
selected.

Reported-by: Simon McVittie <smcv@debian.org>
Closes: #690943
Signed-off-by: Gergely Nagy <algernon@madhouse-project.org>
---
 playercode/mdreg.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/playercode/mdreg.c b/playercode/mdreg.c
index 4018c7e..f158e18 100644
--- a/playercode/mdreg.c
+++ b/playercode/mdreg.c
@@ -82,6 +82,8 @@ void _mm_registeralldrivers(void)
 	_mm_registerdriver(&drv_mac);
 #endif
 
+	_mm_registerdriver(&drv_nos);
+
 	/* Register disk writers */
 	_mm_registerdriver(&drv_raw);
 	_mm_registerdriver(&drv_wav);
@@ -93,8 +95,6 @@ void _mm_registeralldrivers(void)
 #ifndef macintosh
 	_mm_registerdriver(&drv_stdout);
 #endif
-
-	_mm_registerdriver(&drv_nos);
 }
 
 void MikMod_RegisterAllDrivers(void)
